### PR TITLE
systemtest: use manager types to create topics

### PR DESCRIPTION
**Pull request by axw**
Rather than creating topics through Terraform, and Strimzi, create them using managers. We now provision the cluster during initialisation (main_test.go), and each test is responsible for creating (and destroying) its topics.

Port forwarding is now done per test. I found that the port forwarder would eventually get stuck and cause tests to time out -- I think something is not closing connections, but I couldn't get to the bottom of it.

`TestShutdown/Kafka/close` is failing, presumably due to the deadlock that should be fixed by https://github.com/elastic/apm-queue/pull/173 (I haven't confirmed)

### CODE REVIEW

File: systemtest/infra_kafka.go [hunk 1]
```diff
@@ -21,233 +21,203 @@ import (
...
+// deleting them when the test completes.
+func CreateKafkaTopics(ctx context.Context, t testing.TB, topics ...apmqueue.Topic) {
+	manager, err := NewKafkaManager(t)
+	require.NoError(t, err)
+	defer manager.Close()
+	err = manager.CreateTopics(ctx, topics...)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		manager, err := NewKafkaManager(t)
```
Comment (marclop) [inline systemtest/infra_kafka.go:126]: Would it make sense to just create a single kafka manage3r and only close it after the topics are deleted in the `t.Cleanup` function?

File: systemtest/infra_kafka.go [hunk 2]
```diff
@@ -21,233 +21,203 @@ import (
...
+func KafkaCommonConfig(t testing.TB) kafka.CommonConfig {
+	commonConfig := kafka.CommonConfig{
+		Brokers: append([]string{}, kafkaBrokers...),
+		Logger:  logger().Desugar().Named("kafka"),
+	}
+	if len(commonConfig.Brokers) == 0 {
+		brokerAddress := portforwardKafka(t)
+		tlsDialer := &tls.Dialer{
+			// running locally, should be fast
+			NetDialer: &net.Dialer{Timeout: time.Second},
```
Comment (marclop) [inline systemtest/infra_kafka.go:210]: we may need to increase this after #95

File: systemtest/infra_kafka.go [hunk 3]
```diff
@@ -21,233 +21,203 @@ import (
...
+			Config:    &tls.Config{InsecureSkipVerify: true},
+		}
+		commonConfig.Brokers = []string{brokerAddress}
+		commonConfig.Dialer = func(ctx context.Context, network, addr string) (net.Conn, error) {
+			// The advertised broker address is only reachable within
+			// the Kubernetes cluster; replace it with the localhost
+			// port-forwarded address.
+			addr = brokerAddress
```
Comment (marclop) [inline systemtest/infra_kafka.go:218]: ah nice, so this removes the need for the mockdns?

**Review by marclop — APPROVED** [review]

File: systemtest/infra_kafka.go [hunk 4]
```diff
@@ -21,233 +21,203 @@ import (
...
-				cfg.Name,
-				cfg.PortForwardResource,
-				cfg.PortForwardMapping,
-			)
-		})
-		if err != nil {
-			return fmt.Errorf("failed to run port forward: %w", err)
-		}
+
+	// Install Strimzi using Helm.
```
Comment (reakaleek) [inline systemtest/infra_kafka.go:78]: There are some complications in installing the strimzi helm chart multiple times in the same cluster when multiple PRs cause concurrent installations of strimzi.

There is a conflict with ClusterRoles and ClusterRole bindings.

Comment (reakaleek) [reply systemtest/infra_kafka.go:78]: In https://github.com/elastic/apm-queue/pull/170 I avoided this, by removing the strimzi part and installing a permantent strimzi operator that listens on any namespace.

File: systemtest/infra_kafka.go (hunk 1 above)
Comment (axw) [reply systemtest/infra_kafka.go:126]: Yes, that would be better. I'll create two cleanups: one to close, one to delete topics.

File: systemtest/infra_kafka.go (hunk 3 above)
Comment (axw) [reply systemtest/infra_kafka.go:218]: yes, there's only one broker, so we can just ignore the advertised address

File: systemtest/infra_kafka.go (hunk 4 above)
Comment (axw) [reply systemtest/infra_kafka.go]: Thanks, I've remove the Strimzi installation from this PR. Longer term I think we should probably stop relying on Strimzi, and try to use a single Kafka broker pod with ephemeral storage.

File: systemtest/infra_kafka.go (hunk 2 above)
Comment (axw) [reply systemtest/infra_kafka.go:201]: It's working fine at the moment, but let's keep an eye on it and update if needed. I'd update it now but I don't want to wait another 15-20 minutes to merge :smile:

**Pull request merged**